ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm

```diff
@@ -280,6 +293,49 @@ - (void)testShouldHandleVnodeOpEvent {
     XCTAssertEqual(kauthResult, KAUTH_RESULT_DEFER);
 }
 
+- (void)testCurrentProcessWasSpawnedByRegularUser {
+    // Defaults should pass for all tests
+    XCTAssertTrue(CurrentProcessWasSpawnedByRegularUser());
+    MockProcess_Reset();
+
+    // Process tree with all UIDs less than 500 were not spawned by regular users
+    MockProcess_AddContext(context, 1 /*pid*/);
+    MockProcess_SetSelfPid(1);
+    MockProcess_AddCredential(1 /*credentialId*/, 1 /*UID*/);
+    MockProcess_AddProcess(1 /*pid*/, 1 /*credentialId*/, 1 /*ppid*/, "test" /*name*/);
+    XCTAssertFalse(CurrentProcessWasSpawnedByRegularUser());
+    MockProcess_Reset();
+
+    // Test a process with a service UID, valid parent pid, but proc_find fails to find parent pid
+    MockCalls::Clear();
+    MockProcess_AddContext(context, 1 /*pid*/);
+    MockProcess_SetSelfPid(1);
```

Comment:
Consider using something other than 1 for PID here.  My understanding is 1 is typically the root of all process trees, and so the intent of the test might be clearer if you used some other random PID instead (with a parent of 501).